Mathlib/Topology/Order/Basic.lean

```diff
@@ -0,0 +1,2795 @@
+  ⟨hsa, fun b hb =>
+    not_lt.1 fun hba =>
+      have : s ∩ { a | b < a } ∈ f ⊓ 𝓝 a := inter_mem_inf hsf (IsOpen.mem_nhds (isOpen_lt' _) hba)
+      let ⟨_x, ⟨hxs, hxb⟩⟩ := Filter.nonempty_of_mem this
+      have : b < b := lt_of_lt_of_le hxb <| hb hxs
+      lt_irrefl b this⟩
+#align is_lub_of_mem_nhds isLUB_of_mem_nhds
+
+theorem isLUB_of_mem_closure {s : Set α} {a : α} (hsa : a ∈ upperBounds s) (hsf : a ∈ closure s) :
+    IsLUB s a := by
+  rw [mem_closure_iff_clusterPt, ClusterPt, inf_comm] at hsf
+  exact isLUB_of_mem_nhds hsa (mem_principal_self s)
+#align is_lub_of_mem_closure isLUB_of_mem_closure
+
+theorem isGLB_of_mem_nhds :
+    ∀ {s : Set α} {a : α} {f : Filter α}, a ∈ lowerBounds s → s ∈ f → NeBot (f ⊓ 𝓝 a) → IsGLB s a :=
+  @isLUB_of_mem_nhds αᵒᵈ _ _ _
+#align is_glb_of_mem_nhds isGLB_of_mem_nhds
+
+theorem isGLB_of_mem_closure {s : Set α} {a : α} (hsa : a ∈ lowerBounds s) (hsf : a ∈ closure s) :
+    IsGLB s a :=
+  @isLUB_of_mem_closure αᵒᵈ _ _ _ s a hsa hsf
+#align is_glb_of_mem_closure isGLB_of_mem_closure
+
+theorem IsLUB.mem_upperBounds_of_tendsto [Preorder γ] [TopologicalSpace γ] [OrderClosedTopology γ]
+    {f : α → γ} {s : Set α} {a : α} {b : γ} (hf : MonotoneOn f s) (ha : IsLUB s a)
+    (hb : Tendsto f (𝓝[s] a) (𝓝 b)) : b ∈ upperBounds (f '' s) := by
+  rintro _ ⟨x, hx, rfl⟩
+  replace ha := ha.inter_Ici_of_mem hx
+  haveI := ha.nhdsWithin_neBot ⟨x, hx, le_rfl⟩
+  refine' ge_of_tendsto (hb.mono_left (nhdsWithin_mono _ (inter_subset_left s (Ici x)))) _
+  exact mem_of_superset self_mem_nhdsWithin fun y hy => hf hx hy.1 hy.2
+#align is_lub.mem_upper_bounds_of_tendsto IsLUB.mem_upperBounds_of_tendsto
+
+-- For a version of this theorem in which the convergence considered on the domain `α` is as `x : α`
+-- tends to infinity, rather than tending to a point `x` in `α`, see `is_lub_of_tendsto_atTop`
+theorem IsLUB.isLUB_of_tendsto [Preorder γ] [TopologicalSpace γ] [OrderClosedTopology γ] {f : α → γ}
+    {s : Set α} {a : α} {b : γ} (hf : MonotoneOn f s) (ha : IsLUB s a) (hs : s.Nonempty)
+    (hb : Tendsto f (𝓝[s] a) (𝓝 b)) : IsLUB (f '' s) b :=
+  haveI := ha.nhdsWithin_neBot hs
+  ⟨ha.mem_upperBounds_of_tendsto hf hb, fun _b' hb' =>
+    le_of_tendsto hb (mem_of_superset self_mem_nhdsWithin fun _ hx => hb' <| mem_image_of_mem _ hx)⟩
+#align is_lub.is_lub_of_tendsto IsLUB.isLUB_of_tendsto
+
+theorem IsGLB.mem_lowerBounds_of_tendsto [Preorder γ] [TopologicalSpace γ] [OrderClosedTopology γ]
+    {f : α → γ} {s : Set α} {a : α} {b : γ} (hf : MonotoneOn f s) (ha : IsGLB s a)
+    (hb : Tendsto f (𝓝[s] a) (𝓝 b)) : b ∈ lowerBounds (f '' s) :=
+  @IsLUB.mem_upperBounds_of_tendsto αᵒᵈ γᵒᵈ _ _ _ _ _ _ _ _ _ _ hf.dual ha hb
+#align is_glb.mem_lower_bounds_of_tendsto IsGLB.mem_lowerBounds_of_tendsto
+
+-- For a version of this theorem in which the convergence considered on the domain `α` is as
+-- `x : α` tends to negative infinity, rather than tending to a point `x` in `α`, see
+-- `is_glb_of_tendsto_atBot`
```

Comment:
```suggestion
-- For a version of this theorem in which the convergence considered on the domain `α` is as
-- `x : α` tends to negative infinity, rather than tending to a point `x` in `α`, see
-- `isGLB_of_tendsto_atBot`
```